homeassistant/components/todo/__init__.py

```diff
@@ -30,7 +30,7 @@
 ENTITY_ID_FORMAT = DOMAIN + ".{}"
 
 TASK_ITEM_FIELDS = {
-    vol.Required("summary"): cv.string,
+    vol.Required("summary"): vol.All(cv.string, vol.Length(min=1)),
```

Comment:
`summary` is still required in the schema. Should that change?

Reply:
This comment got me thinking about partial update in general.

I've gone ahead and pushed the schemas back into the service directly so we can diverge between create and update if needed. Now they diverge.

For Create, i think it makes sense to stay required.
For Update, the difference is about partial update. I *think* users would want partial update support to make it easier to just change the fields we want to change, without having to read the existing item first. As a result, I went ahead and made this optional and adjusted the test cases. Here are the cases:

todo.update_item:   # Update the summary only, leave status alone
  - uid: 12345
    summary: "foo"

todo.update_item:   # Update the status only, leave summary alone
  - uid: 12345
    status: "completed"


This required updating the data classes parsing and tyying (now slightly less useful since everything is optional across create/udpate cases).

As a consequence, this is now possible, but a no-op, though I think its ok to leave as is.

todo.update_item:
  - summary: "foo"